qlib/utils/time.py

```diff
@@ -120,6 +121,57 @@ def parse(freq: str) -> Tuple[int, str]:
         }
         return _count, _freq_format_dict[_freq]
 
+    @staticmethod
+    def calc_freq_interval(left_frq: str, right_freq: str):
+        """Calculate freq interval; return minutes
+
+        Parameters
+        ----------
+        left_frq: str
+        right_freq: str
+
+        Returns
+        -------
+
+        """
+        minutes_map = {
+            Freq.NORM_FREQ_MINUTE: 1,
+            Freq.NORM_FREQ_DAY: 60 * 24,
+            Freq.NORM_FREQ_WEEK: 7 * 60 * 24,
+            Freq.NORM_FREQ_MONTH: 30 * 7 * 60 * 24,
+        }
+        left_freq = Freq.parse(left_frq)
+        left_minutes = left_freq[0] * minutes_map[left_freq[1]]
+        right_freq = Freq.parse(right_freq)
+        right_minutes = right_freq[0] * minutes_map[right_freq[1]]
+        return left_minutes - right_minutes
+
+    @staticmethod
+    def get_recent_freq(base_freq: str, freq_list: List[str]) -> str:
+        """Get the closest freq to base_freq from freq_list
+
+        Parameters
+        ----------
+        base_freq
+        freq_list
+
+        Returns
+        -------
+
+        """
+        # use the nearest freq greater than 0
+        _freq_minutes = []
+        min_freq = None
+        for _freq in freq_list:
+            _minutes = Freq.calc_freq_interval(base_freq, _freq)
```

Comment:
I think ` _min_delta = Freq.get_min_delta(base_freq, _freq)` will be more readable